### PR TITLE
Added euclidean_gcd [python]

### PR DESCRIPTION
**Pull request by Monal5031**
Added euclidean_gcd [python]

### CODE REVIEW

File: euclidean_gcd/euclidean_gcd.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		first = temp
+	return first 		# When second becomes 0, first becomes gcd of both
+
+def main():
+	first = 36
+	second = 6
+	answer_iterative = euclidean_gcd(first, second)
+	answer_recursive = euclidean_gcd_recursive(first, second)
+	print("GCD of %d and %d is : %d by recursive algo.\n" % (first, second, answer_recursive) )
+	print("GCD of %d and %d is : %d by iterative algo.\n" % (first, second, answer_iterative) )
```
Comment (singhpratyush) [inline euclidean_gcd/euclidean_gcd.py]: Remove `\n` from `print(...)`.

File: euclidean_gcd/euclidean_gcd.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+"""
```
Comment (singhpratyush) [inline euclidean_gcd/euclidean_gcd.py]: Add these comments as function docstring.

File: euclidean_gcd/euclidean_gcd.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	else:
+		return euclidean_gcd(second, first % second )
+
+def euclidean_gcd(first, second):
+	while second != 0:		# Iterate till second becomes zero
+		temp = second 		# Temporary variable to hold value of second which is to be assigned to first later
+		second = first % second
+		first = temp
+	return first 		# When second becomes 0, first becomes gcd of both
+
```
Comment (singhpratyush) [inline euclidean_gcd/euclidean_gcd.py]: Add proper line spacing. See sample code.

File: euclidean_gcd/euclidean_gcd.py (hunk 2 above)
Comment (Monal5031) [inline euclidean_gcd/euclidean_gcd.py]: Both functions had the same param thats why I added it as common on the top.

File: euclidean_gcd/euclidean_gcd.py (hunk 3 above)
Comment (Monal5031) [inline euclidean_gcd/euclidean_gcd.py]: Sorry but I didnt get this one?

File: euclidean_gcd/euclidean_gcd.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+def main():
+	first = 36
+	second = 6
+	answer_iterative = euclidean_gcd(first, second)
+	answer_recursive = euclidean_gcd_recursive(first, second)
+	print("GCD of %d and %d is : %d by recursive algo." % (first, second, answer_recursive) )
+	print("GCD of %d and %d is : %d by iterative algo." % (first, second, answer_iterative) )
```
Comment (singhpratyush) [inline euclidean_gcd/euclidean_gcd.py]: Remove the single space in the end please.

File: euclidean_gcd/euclidean_gcd.py (hunk 2 above)
Comment (singhpratyush) [inline euclidean_gcd/euclidean_gcd.py]: Add that to the `euclidean_gcd` function. Use standard method.

File: euclidean_gcd/euclidean_gcd.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+ 	:param first --> First number
+ 	:param second --> Second number
+	"""
+	if second == 0:
+		return first        # First becomes gcd if second becomes zero
+	else:
+		return euclidean_gcd(second, first % second )
+
+
+def euclidean_gcd(first, second):
```
Comment (singhpratyush) [inline euclidean_gcd/euclidean_gcd.py]: Please standardize the docstrings -
```python
def euclidean_gcd(first, second):
    """
    Calculates GCD of two nummbers using Euclidean Algorithm
    :param first: First number
    :param second: Second number
    :return: GCD of the numbers
    """
    ...
```

File: euclidean_gcd/euclidean_gcd.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
+#There are two implementations: Recursive (euclidean_gcd_recursive) and Non-Recursive (euclidean_gcd)
```
Comment (singhpratyush) [inline euclidean_gcd/euclidean_gcd.py]: If a python file starts with a comment, it should be a block comment (`"""`).

Comment (Monal5031) [inline euclidean_gcd/euclidean_gcd.py]: Okay got it.

File: euclidean_gcd/euclidean_gcd.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+	"""
+    Calculates GCD of two numbers using Euclidean Recursive Algorithm
+    :param first: First number
+    :param second: Second number
+    :return: GCD of the numbers
+    """
+	if second == 0:
+		return first        # First becomes gcd if second becomes zero
+	else:
+		return euclidean_gcd(second, first % second )
```
Comment (singhpratyush) [inline euclidean_gcd/euclidean_gcd.py]: Also this one - extra space before `)`.

**Comment by singhpratyush** [comment]
I recommend that you use some linting tool like `pep8` before making changes -
```bash
pip install pep8
pep8 /path/to/python/file.py
```

File: euclidean_gcd/euclidean_gcd.py [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+	return first 		# When second becomes 0, first becomes gcd of both
+
+
+def main():
+	first = 36
+	second = 6
+	answer_iterative = euclidean_gcd(first, second)
+	answer_recursive = euclidean_gcd_recursive(first, second)
+	print("GCD of %d and %d is : %d by recursive algo." % (first, second, answer_recursive))
+	print("GCD of %d and %d is : %d by iterative algo." % (first, second, answer_iterative))
```
Comment (Monal5031) [inline euclidean_gcd/euclidean_gcd.py]: Here I exceed 79 chars any suggestion that how can I fix this one?

Comment (singhpratyush) [inline euclidean_gcd/euclidean_gcd.py]: Add line breaks
```python
    print("GCD of %d and %d is : %d by recursive algo." % (first, second,
                                                           answer_recursive))
```

**Pull request merged**